### PR TITLE
pkbdf2

### PR DESCRIPTION
**Pull request by tannernelson**

### CODE REVIEW

File: Sources/PBKDF2/PBKDF2.swift [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import Essentials
+import HMAC
+
+public enum PBKDF2Error: Error {
+    case cannotIterate(times: Int)
+    case cannotDeriveFromKey(Bytes)
+    case cannotDeriveFromSalt(Bytes)
+    case keySizeTooBig(UInt)
+}
+/*
```
Comment (LoganWright) [inline Sources/PBKDF2/PBKDF2.swift]: Is this supposed to be commented out?

**Comment by tannernelson** [comment]
Closing this for now.

**Comment by robertjpayne** [comment]
Curious why this was closed? We need PBKDF2 to do RethinkDB SCRAM auth, would be great to have it!

**Comment by tannernelson** [comment]
@robertjpayne there wasn't enough time to clean up the API and fully test. We will add back at some point #13

**Pull request closed without merging**